File: app/src/main/java/com/egecius/electriccars/room/CarDao.java
package com.egecius.electriccars.room;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface CarDao {

    @Query("SELECT * FROM cars")
    List<CarRoom> getAllCars();

    @Insert (onConflict = OnConflictStrategy.REPLACE)
    void insertCar(CarRoom car);

    @Query("DELETE FROM Cars")
    void deleteAllCars();

}

File: app/src/main/java/com/egecius/electriccars/room/CarDao.kt
package com.egecius.electriccars.room

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface CarDao {

    @Query("SELECT * FROM cars")
    suspend fun loadAllCars(): List<Car>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertCar(car: Car)

    @Query("DELETE FROM Cars")
    fun deleteAllCars()

}
